CONTRIBUTING.rst

```diff
@@ -254,6 +255,15 @@ Guidelines
           # function not for directed graphs *or* for multigraphs
           pass
 
+* Adding new default and extra dependencies should be done only with careful
+  consideration.  New default dependencies should be widely-used in the community
+  and have high potential for wide-spread use in NetworkX.  They should also be
+  pure Python and easy to install.  Extra dependencies should also be easy to
+  install and widely-used, but may have low potential for wide-spread use in
+  NetworkX.  Typically, new default dependencies would first be added as new
+  extra dependencies.  Default dependencies should be added to
```

Comment:
I was thinking that, in general, new packages would be first added as an extra dependency (i.e., added to ``requirements/extra.txt``).  If it was used more generally, it might eventually be changed to a default dependency (i.e., moved from ``requirements/extra.txt`` to ``requirements/default.txt``).

Reply:
So, I think your language is fine.  I will add ``requirements/extra.txt`` and ``requirements/default.txt`` to make my intention explicit.